batch/batch/worker/worker.py

```diff
@@ -1310,7 +1310,9 @@ async def setup_io(self):
                     project=PROJECT,
                     instance_name=NAME,
                     name=f'batch-disk-{uid}',
-                    compute_client=worker.compute_client,
+                    compute_client=aiogoogle.ComputeClient(
+                        PROJECT, credentials=aiogoogle.Credentials.from_file('/worker-key.json')
+                    ),
```

Comment:
The Disk assumes it doesn't own the ComputeClient. I think the most future-proof move here is to have the Disk create a ComputeClient itself inside the constructor. It knows the project and the Credentials come from an expected place. Then the Disk needs a close method. That method needs to be called when the Disk is no longer needed. I think that's `DockerJob.cleanup`

Reply:
ah yes you're right. Haven't started on my coffee yet.